core/src/testUtils/groovy/nu/yona/server/test/User.groovy

```diff
@@ -50,7 +50,7 @@ class User
 		{
 			this.nickname = json.nickname
 
-			this.buddies = (json._embedded?.buddies) ? json._embedded.buddies.collect{new Buddy(it)} : []
+			this.buddies = (json._embedded?.buddies && json._embedded.buddies._embedded) ? json._embedded.buddies._embedded.buddies.collect{new Buddy(it)} : []
```

Comment:
The work in this PR is good, but it is not addressing the JIRA issue yet, so I prefer to keep it open. Once YD-172 is done, this ticket can be taken on again to add the requested links.